lib/helper/firefoxProfileCreator.py

```diff
@@ -36,10 +36,18 @@ def get_firefox_profile(self, prefs={}, cookies_settings={}, profile_files_setti
 
     def _create_firefox_profile(self):
         tmp_profile_dir = tempfile.mkdtemp(prefix='firefoxprofile_')
+        quoted_firefox_cmd = '"%s"' % self.firefox_cmd.strip()
+
         logger.info('Creating Profile: {}'.format(tmp_profile_dir))
-        os.system('{} --profile {} -silent'.format(self.firefox_cmd, tmp_profile_dir))
-        self._firefox_profile_path = tmp_profile_dir
-        logger.info('Creating Profile success: {}'.format(self._firefox_profile_path))
+        status = os.system('{} --profile {} -silent'.format(quoted_firefox_cmd, tmp_profile_dir))
```

Comment:
if we replace `os.system` by `subprocess.check_call`, then it will run command and wait for command to complete. If the return code was not zero then raise `CalledProcessError`.